app/models/conference.rb

```diff
@@ -81,6 +82,21 @@ def user_registered? user
   end
 
   ##
+  # Delete all EventSchedules that are not in the hours range
+  # After the conference has been successfully updated
+  def delete_event_schedules
+    if self.start_hour_changed? || self.end_hour_changed?
+      event_schedules = EventSchedule.select do |e|
```

Comment:
@siddhantbajaj I mean adding a `has_many thought` relation in program to be able to call it like 

``` Ruby
conference.program.event_schedules.select do |event_schedule|
```

http://guides.rubyonrails.org/association_basics.html#the-has-many-through-association